### DISCUSSION TITLE
How to add custom variant before :active selector

### DISCUSSION THREAD

**Discussion by stijns96** [tailwindlabs/tailwindcss#12797] [Help]
Hi guys,

I'm trying to add my custom `hocus:` variant to my button together with the default `active:` selector. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:active), the active state should come after the hover.

_To style links appropriately, put the :active rule after all other link-related rules, as defined by the LVHA-order: :link — :visited — :hover — :active._

With the default hover it's working fine.

**Custom (hocus)**
![image](https://github.com/tailwindlabs/tailwindcss/assets/47820758/f0074cc1-10ec-4d76-8de0-ba6ac4536f9c)

**Hover**
![image](https://github.com/tailwindlabs/tailwindcss/assets/47820758/094c7073-776b-4a5e-8e31-336fec490f02)

Is there any option to set the position of custom variants?

Thanks!

**Comment by wongjn** [comment]
I don't believe there is any option to set the position of custom variants relative to other variants.

However, for your particular situation, you could consider re-declaring the `active:` variant after your `hocus:`, which should make the `:active` rules come after your `:hover/:focus`:

```js
require('tailwindcss/plugin')(({ addVariant }) => {
  addVariant('hocus', ['&:hover', '&:focus']);
  addVariant('active', ['&:active']);
})
```

[See this Tailwind Play](https://play.tailwindcss.com/abDecvXIhI?file=config) as an example.

**Reply by stijns96** [reply]
Mhh, I'll try it asap!

@adamwathan is it somehow possible to add a specificity kind of setting to this or is this situation to rare to put any effort in?
